test/framework/src/test/java/org/elasticsearch/transport/nio/SimpleNioTransportTests.java

```diff
@@ -87,15 +95,26 @@ protected SocketEventHandler getSocketEventHandler() {
     }
 
     @Override
-    protected MockTransportService build(Settings settings, Version version, ClusterSettings clusterSettings, boolean doHandshake) {
+    protected MockTransportService build(
+            Settings settings, Version version, ClusterSettings clusterSettings, boolean doHandshake, Consumer<NioChannel> onChannelOpen) {
         settings = Settings.builder().put(settings)
             .put(TcpTransport.PORT.getKey(), "0")
             .build();
-        MockTransportService transportService = nioFromThreadPool(settings, threadPool, version, clusterSettings, doHandshake);
+        MockTransportService transportService =
+                nioFromThreadPool(settings, threadPool, version, clusterSettings, doHandshake, onChannelOpen);
         transportService.start();
         return transportService;
     }
 
+    @Override
+    protected void close(NioChannel nioChannel) {
+        try {
+            nioChannel.getCloseFuture().awaitClose();
```

Comment:
Out of curiosity - this reads weird - we wait on close but not actively close? is that OK?

Reply:
Oh, that was silly. I pushed d69c92e03cb47fedc622d7decaea292a63ea1a3b.